background/background.js

```diff
@@ -147,12 +136,22 @@ prefs.initializing.then(() => {
   refreshAllIcons();
 });
 
-navigatorUtil.onUrlChange(({tabId, frameId, transitionQualifiers}, type) => {
-  if (type === 'committed' && !frameId) {
+navigatorUtil.onUrlChange(({tabId, frameId, transitionQualifiers, url}, type) => {
+  if (type !== 'committed') {
+    msg.sendTab(tabId, {method: 'urlChanged'}, {frameId})
+      .catch(msg.ignoreError);
+  } else if (!frameId) {
+    if (url.includes('.user.') &&
+        /^(https?|file|ftps?):/.test(url) &&
+        /\.user\.(css|styl)$/.test(url.split(/[#?]/, 1)[0]) &&
+        !`${tabIconFor(tabId).url}`.startsWith(URLS.installUsercss)) {
```

Comment:
I think it's a bad idea to mix the installation logic with the tab icon logic. Let's split out tab icon logic to a different file e.g. `background/icon-manager.js`.

Reply:
This is something you can do later. As for me, I'm fine with simply renaming the variable to tabData.